test/server/listener_manager_impl_test.cc

```diff
@@ -1179,6 +1179,16 @@ TEST_F(ListenerManagerImplWithRealFiltersTest, MultipleFilterChainsWithServerNam
   server_names = ssl_socket->dnsSansLocalCertificate();
   EXPECT_EQ(server_names.size(), 2);
   EXPECT_EQ(server_names.front(), "*.example.com");
+
+  // TLS client with wildcard SNI match - using 3nd filter chain.
+  filter_chain = findFilterChain("www.wildcard.com", true, "tls", true, {});
+  ASSERT_NE(filter_chain, nullptr);
+  EXPECT_TRUE(filter_chain->transportSocketFactory().implementsSecureTransport());
+  transport_socket = filter_chain->transportSocketFactory().createTransportSocket();
+  ssl_socket = dynamic_cast<Ssl::SslSocket*>(transport_socket.get());
+  server_names = ssl_socket->dnsSansLocalCertificate();
+  EXPECT_EQ(server_names.size(), 2);
```

Comment:
test nit: Should you be checking the 3rd server name here?

Reply:
No, this check is verifying that the 3rd filter chain was served by looking at the the certificate (which has 2 SANs and `*.example.com` is the first of them).

Reply:
OK got it, thanks.